tracer/src/Datadog.Trace/Configuration/GitMetadataTagsProvider.cs

```diff
@@ -93,11 +95,12 @@ private bool TryGetGitTagsFromSourceLink([NotNullWhen(true)] out GitMetadata? re
         {
             // Cannot determine the entry assembly. This may mean this method was called too early.
             // Return false to indicate that we should try again later.
-
+            // We'll try up to 100 times, but if a span is active, we'll give up immediately, as that means
+            // the application is already fully up and running and we're not going to be able to retrieve the entry assembly.
             var nbTries = Interlocked.Increment(ref _tryCount);
-            if (nbTries > 100)
+            if (nbTries > 100 || _scopeManager.Active?.Span != null)
```

Comment:
If `_scopeManager.Active?.Span` is not null, shouldn't you also set `nbTries` to 100? Otherwise, the other threads (for instance, RCM) will keep trying.